skllm/llm/gpt/mixin.py

```diff
@@ -66,8 +66,14 @@ def _set_keys(self, key: Optional[str] = None, org: Optional[str] = None) -> Non
         """
         Set the OpenAI key and organization.
         """
-        self.openai_key = key
-        self.openai_org = org
+            
+        self.key = key
+        self.org = org
+        
+        if self.key is None:
+            self.key = self._get_openai_key()
```

Comment:
To be honest, I would not store the key from the global config inside the estimator. First of all, the idea of the global config is that the key can be dynamically propagated into all estimators (unless the estimator was provided with its own key) even after instantiation. 

In addition, when the key is stored globally, the object can be safely pickled, whereas otherwise it would be necessary to remove the key first.

In other words, there should be support for per-estimator keys for cases when this is absolutely needed, but this should not be the default behavior.

Reply:
Agreed, I removed it.